File: 539. Minimum Time Difference/_539.py
class Solution:
    def findMinDifference(self, timePoints) -> int:
        timelist = list()
        for i in timePoints:
            h,m = i.split(':')
            timelist.append(int(h)*60+int(m))

        timelist.sort()
        minvalue = timelist[1]-timelist[0]
        for i in range(2, len(timelist)):    
            minvalue = min(timelist[i]-timelist[i-1], minvalue)

        return min(minvalue, 24*60+timelist[0]-timelist[-1])

    def findMinDifference1(self, timePoints):
        t = sorted(int(t[:2]) * 60 + int(t[-2:]) for t in timePoints)
        t.append(t[0] + 1440)
        return min(b - a for a, b in zip(t, t[1:]))

File: 539. Minimum Time Difference/539.go
package leetcode

import(
	"strings"
	"strconv"
	"sort"
)

func findMinDifference(timePoints []string) int {
	timeList := make([]int, 0, len(timePoints))
	for _,v := range timePoints{
		hm := strings.Split(v, ":")
		hour,_ := strconv.Atoi(hm[0])
		minute,_ := strconv.Atoi(hm[1])
		timeList = append(timeList, hour*60+minute)
	}

	sort.Ints(timeList)

	minvalue := timeList[1]-timeList[0]

	for i:=2;i<len(timeList);i++{
		tmp := timeList[i]-timeList[i-1]
		if tmp<minvalue{
			minvalue = tmp
		}
	}

	tmp:=60*24+timeList[0]-timeList[len(timeList)-1]
	if tmp<minvalue{
		return tmp
	}
	return minvalue
}